### PR TITLE
Fix function deprecation

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
BinaryBuilder is popping up with the following deprecation warnings in its CI, which seem to be caused by GitHub.jl:
Screen Shot 2022-01-10 at 14 14 27

### CODE REVIEW

**Comment by fredrikekre** [comment]
It would be better to add a dependency on the URIs package instead of reaching for the URI type from withing HTTP.

**Comment by jeremiahpslewis** [comment]
@fredrikekre Not a problem, but is that change maybe a separate decision / concern? `HTTP.URI` is used all over the package already?

https://github.com/JuliaWeb/GitHub.jl/blob/8d734e4a3d1fa82edaf2bde5634a2bea9714c6c8/test/ghtype_tests.jl#L55

**Comment by fredrikekre** [comment]
Fair. Does the lowest HTTP compat in this package have the new URI constructor?

**Comment by jeremiahpslewis** [comment]
Feature was added to URIs.jl here: https://github.com/JuliaWeb/URIs.jl/pull/12, e.g. URIs v1.1 https://github.com/JuliaRegistries/General/pull/25219

This is supported by HTTP.jl v0.9.7 and later, so we would have to bump the compat.

**Comment by fredrikekre** [comment]
Okay, adding URIs as a dep might be easier then since it is more explicit (you don't need to remove the uses of `HTTP.URI` even if you do that, but might be a nice clean up.)

**Comment by jeremiahpslewis** [comment]
In order to merge this PR, we will need to require a version of HTTP.jl which supports URIs.jl v1.1, which is only a subset of HTTP.jl 0.9; as a result we should drop support for 0.8 and maybe make the HTTP bounds on 0.9, e.g. 0.9.7 explicit.

**Comment by fredrikekre** [comment]
The requirement of URIs >= 1.1 would implicitly require a correct HTTP version, so changing HTTP compat is not actually necessary.

File: Project.toml [hunk 1]
```diff
@@ -10,9 +10,11 @@ JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
 MbedTLS = "739be429-bea8-5141-9913-cc70e7f3736d"
 Sockets = "6462fe0b-24de-5631-8697-dd941f90decc"
 SodiumSeal = "2133526b-2bfb-4018-ac12-889fb3908a75"
+URIs = "5c2747f8-b7ea-4ff2-ba2e-563bfd36b1d4"
 
 [compat]
-HTTP = "0.8, 0.9"
+HTTP = "0.9.7"
+URIs = "1.1, 1.2, 1.3"
```
Comment (fredrikekre) [inline Project.toml:17]: Equivalent to
```suggestion
URIs = "1.1"
```

**Comment by jeremiahpslewis** [comment]
@fredrikekre Do we want to be explicit about dropping HTTP 0.8? Or is there an advantage to leaving it in compat if it gets ruled out during installation?

**Comment by fredrikekre** [comment]
Yea, might as well.

File: Project.toml [hunk 2]
```diff
@@ -10,9 +10,11 @@ JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
 MbedTLS = "739be429-bea8-5141-9913-cc70e7f3736d"
 Sockets = "6462fe0b-24de-5631-8697-dd941f90decc"
 SodiumSeal = "2133526b-2bfb-4018-ac12-889fb3908a75"
+URIs = "5c2747f8-b7ea-4ff2-ba2e-563bfd36b1d4"
 
 [compat]
-HTTP = "0.8, 0.9"
+HTTP = "0.9"
+URIs = "1.1, 1.3"
```
Comment (fredrikekre) [inline Project.toml:17]:
```suggestion
URIs = "1.1"
```

Comment (jeremiahpslewis) [reply Project.toml:17]: This pins URIs to 1.1, even though 1.2 and 1.3 have been released. What is the advantage of this?

Comment (jeremiahpslewis) [reply Project.toml:17]: Gotcha. Reread the docs, thanks!

**Review by fredrikekre — APPROVED** [review]

**Pull request merged**